File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/service/cadastros/BancoService.java
package com.t2ti.fenix.service.cadastros;

import java.util.List;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.Query;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.t2ti.fenix.model.cadastros.Banco;
import com.t2ti.fenix.model.transiente.Filtro;
import com.t2ti.fenix.repository.cadastros.BancoRepository;

@Service
public class BancoService {

	@Autowired
	private BancoRepository repository;
	
	@PersistenceContext
    private EntityManager entityManager;	
	
	public List<Banco> consultarLista() {
		return repository.findAll();
	}

	@SuppressWarnings("unchecked")
	public List<Banco> consultarLista(Filtro filtro) {
		String sql = "select * from banco where " + filtro.getWhere();
		Query query = entityManager.createNativeQuery(sql, Banco.class);
		List<Banco> lista = (List<Banco>) query.getResultList();
		return lista;
	}
	
	public Banco consultarObjeto(Integer id) {
		return repository.findById(id).get();
	}
	
	public Banco salvar(Banco objeto) {
		return repository.save(objeto);
	}
		
	public void excluir(Integer id) {
		Banco objeto = new Banco();
		objeto.setId(id);
		repository.delete(objeto);
	}
	
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/php/fenix/src/service/BancoService.php
<?php

require './src/config/doctrine.php';
require './src/model/Banco.php';

class BancoService
{
    public static function consultarLista()
    {
        $gerente = GerenteConexao::getInstance();        
        $bancoRepository = $gerente->entityManager->getRepository('Banco');
        return $bancoRepository->findAll();
    }

    public static function consultarListaFiltroValor(string $campo, string $valor)
    {
        $gerente = GerenteConexao::getInstance();   
        $sql = "SELECT b FROM Banco b WHERE b.$campo like :valor";     
        $query = $gerente->entityManager->createQuery($sql);
        $query->setParameter("valor", "%$valor%");
        return $query->getResult();
    }

    public static function consultarObjeto(int $id)
    {
        $gerente = GerenteConexao::getInstance();        
        return $gerente->entityManager->find('Banco', $id);
    }

    public static function salvar($banco)
    {
        $gerente = GerenteConexao::getInstance();        
        $gerente->entityManager->persist($banco);
        $gerente->entityManager->flush();
    }

    public static function excluir($banco)
    {
        $gerente = GerenteConexao::getInstance();        
        $gerente->entityManager->remove($banco);
        $gerente->entityManager->flush();
    }

}